### PR TITLE
Update module definitions and resolve `Module swapped` exception

### PR DESCRIPTION
**Pull request by Tkael**
- Add EDDBIDs to newer module definitions
- Add missing modules from https://github.com/EDCD/FDevIDs/pull/33/files
- Improve reporting of missing module definitions
- Refactor and harden ShipMonitor.cs against exceptions like "An item with the same key has already been added." Ref. https://rollbar.com/EDCD/EDDI/items/15048/
- Enhance `Module swapped` unit testing.
- Implement suggestions from code analysis.

### CODE REVIEW

**Comment by Hoodathunk** [comment]
Please add back the module sorting after swapping, etc to ensure the associated json data is properly ordered for readability.

**Comment by Tkael** [comment]
@Hoodathunk that's what the `sortCompartments` and `sortHardpoints` methods are for, unless I missed applying them somewhere.

**Comment by Hoodathunk** [comment]
OK, my bad. Hard to follow all the red & green. Had to download the branch to better identify the changes.
